### PR TITLE
Fix docs for `DataFrame$explode()` and `LazyFrame$explode()`

### LINKED ISSUE #461: `LazyFrame$explode` and `DataFrame$explode` seems wrong
While looking at the website, I noticed that the behavior of explode is different from the description.
I think the documentation or implementation and testing are wrong.

```r
df = pl$DataFrame(
  letters = c("aa", "aa", "bb", "cc"),
  numbers = list(1, c(2, 3), c(4, 5), c(6, 7, 8)),
  numbers_2 = list(0, c(1, 2), c(3, 4), c(5, 6, 7)) # same structure as numbers
)
df
#> shape: (4, 3)
#> ┌─────────┬─────────────────┬─────────────────┐
#> │ letters ┆ numbers         ┆ numbers_2       │
#> │ ---     ┆ ---             ┆ ---             │
#> │ str     ┆ list[f64]       ┆ list[f64]       │
#> ╞═════════╪═════════════════╪═════════════════╡
#> │ aa      ┆ [1.0]           ┆ [0.0]           │
#> │ aa      ┆ [2.0, 3.0]      ┆ [1.0, 2.0]      │
#> │ bb      ┆ [4.0, 5.0]      ┆ [3.0, 4.0]      │
#> │ cc      ┆ [6.0, 7.0, 8.0] ┆ [5.0, 6.0, 7.0] │
#> └─────────┴─────────────────┴─────────────────┘

# it doesn't change anything if the input is not a list-column
df$explode("letters")
#> shape: (8, 3)
#> ┌─────────┬─────────────────┬─────────────────┐
#> │ letters ┆ numbers         ┆ numbers_2       │
#> │ ---     ┆ ---             ┆ ---             │
#> │ str     ┆ list[f64]       ┆ list[f64]       │
#> ╞═════════╪═════════════════╪═════════════════╡
#> │ a       ┆ [1.0]           ┆ [0.0]           │
#> │ a       ┆ [1.0]           ┆ [0.0]           │
#> │ a       ┆ [2.0, 3.0]      ┆ [1.0, 2.0]      │
#> │ a       ┆ [2.0, 3.0]      ┆ [1.0, 2.0]      │
#> │ b       ┆ [4.0, 5.0]      ┆ [3.0, 4.0]      │
#> │ b       ┆ [4.0, 5.0]      ┆ [3.0, 4.0]      │
#> │ c       ┆ [6.0, 7.0, 8.0] ┆ [5.0, 6.0, 7.0] │
#> │ c       ┆ [6.0, 7.0, 8.0] ┆ [5.0, 6.0, 7.0] │
#> └─────────┴─────────────────┴─────────────────┘
```

<https://github.com/pola-rs/r-polars/blob/0e730bccb48a0126283235cbe3834f57ae678fcd/R/dataframe__frame.R#L1628-L1649>

The str type column `letters` is decomposed by explode.

### PR DESCRIPTION
**Pull request by etiennebacher**
Close #461

### CODE REVIEW

File: R/dataframe__frame.R [hunk 1]
```diff
@@ -1640,9 +1640,6 @@ DataFrame_glimpse = function(..., return_as_string = FALSE) {
 #' # explode a single column, append others
 #' df$explode("numbers")
 #'
-#' # it doesn't change anything if the input is not a list-column
-#' df$explode("letters")
```
Comment (eitsupi) [inline R/dataframe__frame.R:1644]: I think this surprising behavior should be included in the example.

**Review by eitsupi — CHANGES_REQUESTED** [review]
Thanks for working on this.

File: R/dataframe__frame.R (hunk 1 above)
Comment (etiennebacher) [reply R/dataframe__frame.R]: done

**Review by eitsupi — APPROVED** [review]

**Comment by eitsupi** [comment]
I think this behavior needs to be tested as the behavior may change upstream.
https://github.com/pola-rs/r-polars/blob/d6ac3825c55973da0ddd1e6d82e5b06601b062b4/tests/testthat/test-lazy.R#L634-L639

@etiennebacher Could you take a look at this? Thanks!

**Pull request merged**